Source/santactl/Commands/SNTCommandMetrics.m

```diff
@@ -0,0 +1,180 @@
+///
+///    Unless required by applicable law or agreed to in writing, software
+///    distributed under the License is distributed on an "AS IS" BASIS,
+///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///    See the License for the specific language governing permissions and
+///    limitations under the License.
+
+#import <Foundation/Foundation.h>
+#import <MOLXPCConnection/MOLXPCConnection.h>
+
+#import "Source/common/SNTCommonEnums.h"
+#import "Source/common/SNTConfigurator.h"
+#import "Source/common/SNTXPCControlInterface.h"
+#import "Source/santactl/SNTCommand.h"
+#import "Source/santactl/SNTCommandController.h"
+
+@interface SNTCommandMetrics : SNTCommand <SNTCommandProtocol>
+@end
+
+@implementation SNTCommandMetrics
+
+REGISTER_COMMAND_NAME(@"metrics")
+
++ (BOOL)requiresRoot {
+  return NO;
+}
+
++ (BOOL)requiresDaemonConn {
+  return YES;
+}
+
++ (NSString *)shortHelpText {
+  return @"Show Santa metric information.";
+}
+
++ (NSString *)longHelpText {
+  return (@"Provides metrics about Santa's operation while it's running.\n"
+          @"  Use --json to output in JSON format");
+}
+
+- (NSString *)stringFromMetricFormat:(SNTMetricFormatType)format {
+  switch (format) {
+    case SNTMetricFormatTypeMonarchJSON: return @"monarchjson";
+    case SNTMetricFormatTypeRawJSON: return @"rawjson";
+    default: return @"Unknown Metric Format";
+  }
+}
+
+- (NSDictionary *)normalize:(NSDictionary *)metrics {
+  NSMutableDictionary *mutableMetrics = [metrics mutableCopy];
+
+  id formatter;
+
+  if (@available(macOS 10.13, *)) {
+    NSISO8601DateFormatter *isoFormatter = [[NSISO8601DateFormatter alloc] init];
+
+    isoFormatter.formatOptions =
+      NSISO8601DateFormatWithInternetDateTime | NSISO8601DateFormatWithFractionalSeconds;
+    formatter = isoFormatter;
+  } else {
+    NSDateFormatter *localFormatter = [[NSDateFormatter alloc] init];
+    [localFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"];
+    [localFormatter setTimeZone:[NSTimeZone timeZoneWithName:@"UTC"]];
+    formatter = localFormatter;
+  }
+
+  for (NSString *metricName in mutableMetrics[@"metrics"]) {
+    NSMutableDictionary *metric = mutableMetrics[@"metrics"][metricName];
+
+    for (NSString *field in metric[@"fields"]) {
+      NSMutableArray<NSMutableDictionary *> *values = metric[@"fields"][field];
+
+      [values enumerateObjectsUsingBlock:^(id object, NSUInteger index, BOOL *stop) {
+        values[index][@"created"] = [formatter stringFromDate:values[index][@"created"]];
+        values[index][@"last_updated"] = [formatter stringFromDate:values[index][@"last_updated"]];
+      }];
+    }
+  }
+
+  return mutableMetrics;
+}
+
+- (void)runWithArguments:(NSArray *)arguments {
+  __block NSDictionary *metrics;
+
+  dispatch_group_t group = dispatch_group_create();
+  dispatch_group_enter(group);
+
+  [[self.daemonConn remoteObjectProxy] metrics:^(NSDictionary *exportedMetrics) {
```

Comment:
Add a unit test to verify that output work as intended (similar to SNTCommandFileInfoTest)

Reply:
Added tests in the latest commit.